pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go

```diff
@@ -239,6 +250,70 @@ func (r *ReconcileQuarksSecret) createRSASecret(ctx context.Context, qsec *qsv1a
 	return r.createSecrets(ctx, qsec, secret)
 }
 
+// renderSecret uses the specified templating engine to draw the secret data
+func (r *ReconcileQuarksSecret) renderSecret(ctx context.Context, qsec *qsv1a1.QuarksSecret) (map[string]string, error) {
+	res := map[string]string{}
+	if qsec.Spec.TemplateType == "" {
+		return res, errors.New("templatedConfig needs a templateType to be specified. E.g. helm")
+	}
+
+	// Interpolate the given values, and retrieve the secret contents to fill our config with
+	for k, v := range qsec.Spec.TemplateValues {
+		fields := strings.Split(v.(string), ".")
```

Comment:
I don't think there is a reason to use a string of two combined values, instead of atomic fields

Reply:
so if I got it right, would be something the lines of: 
```
template:
    secrets:
    - name: "secretname"
      field: "password"
```
?

Reply:
I  re-used the existing SecretReference